src/Microsoft.AspNet.Mvc.Core/ModelBinding/CollectionModelBinder.cs

```diff
@@ -84,14 +85,61 @@ public class CollectionModelBinder<TElement> : IModelBinder
                 validationNode: result?.ValidationNode);
         }
 
-        // Called when we're creating a default 'empty' model for a top level bind.
-        protected virtual object CreateEmptyCollection()
+        /// <inheritdoc />
+        /// <remarks>Called when we're creating a default 'empty' model for a top level bind.</remarks>
+        public virtual object CreateEmptyCollection(Type targetType)
         {
-            return new List<TElement>();
+            if (targetType.IsAssignableFrom(typeof(List<TElement>)))
+            {
+                // Simple case such as ICollection<TElement>, IEnumerable<TElement> and IList<TElement>.
+                return new List<TElement>();
+            }
+
+            return CreateInstance(targetType);
+        }
+
+        /// <summary>
+        /// Create an instance of <paramref name="targetType"/>.
+        /// </summary>
+        /// <param name="targetType"><see cref="Type"/> of the model.</param>
+        /// <returns>An instance of <paramref name="targetType"/>.</returns>
+        protected object CreateInstance(Type targetType)
+        {
+            try
+            {
+                return Activator.CreateInstance(targetType);
+            }
+            catch (COMException)
+            {
+                // COM oddity part 1.
+            }
+            catch (InvalidComObjectException)
+            {
+                // COM oddity part 2.
+            }
+            catch (MemberAccessException)
+            {
+                // Type is abstract, has no empty constructor, or constructor is not accessible.
+            }
+            catch (NotSupportedException)
+            {
+                // Activator does not support given type.
+            }
+            catch (TargetInvocationException)
+            {
+                // Constructor threw an Exception.
+            }
+            catch (TypeLoadException)
+            {
+                // Invalid type.
+            }
+
+            return null;
```

Comment:
If we're really determined to just null out here, then we should handle all exceptions. Documented exceptions thrown from a method in the CLR aren't a contract and can change.